public/components/security/roles-mapping/components/rule-editor.tsx

```diff
@@ -415,9 +463,40 @@ export const RuleEditor = ({ save, initialRule, isLoading, isReserved, internalU
       <EuiFlexGroup style={{ marginTop: 6 }}>
         <EuiFlexItem></EuiFlexItem>
         <EuiFlexItem grow={false}>
-          <EuiButton disabled={isReserved} isLoading={isLoading} fill onClick={() => saveRule()}>
+          <WzButtonPermissions
+            buttonType='default'
+            permissions={[
+              ...(typeof ruleID !== 'undefined'
+                ? /* Add the permission when creating a security rule */
+                  [{ action: 'security:create', resource: '*:*:*' }]
+                : []),
+              /* TODO:
+                The required permissions depends on the form. Editing could need the security:update
+                action or security:delete depending on some roles are being added or are removed. To
+                be more exact we should detect when the role is being added or removed.
+              */
+              {
+                action: 'security:update',
+                resource:
+                  typeof ruleID !== 'undefined'
+                    ? `rule:id:${ruleID}`
+                    : 'rule:id:*',
+              },
+              {
+                action: 'security:delete',
+                resource:
+                  typeof ruleID !== 'undefined'
+                    ? `rule:id:${ruleID}`
+                    : 'rule:id:*',
+              },
+            ]}
```

Comment:
This component is being used when editing or creating, and asks for the same permissions for both cases, I think that the "security:create" permission would not be necessary when editing, the "security:delete" permission would not be necessary when creating and the "security:update" permission would only be necessary when editing roles.

Reply:
The `security:create` permission is only applying when creating the role mapping.

I will remove the `security:delete` permission when creating the role mapping.

I will research if we could get a more fine-grain permission requirements.

Reply:
I did some changes here: https://github.com/wazuh/wazuh-kibana-app/pull/5551/commits/ab294761b3cc18b66984aa8093aaecc79a5a4655